pkg/cache/memory/memory_test.go

```diff
@@ -0,0 +1,147 @@
+package memory
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	logContext "github.com/trufflesecurity/trufflehog/v3/pkg/context"
+)
+
+func TestCache(t *testing.T) {
+	c := New()
+
+	// Test set and get.
+	c.Set("key1", true)
+	v, ok := c.Get("key1")
+	if !ok {
+		t.Fatalf("Unexpected value for key1: %v, %v", v, ok)
+	}
+
+	// Test the count.
+	if c.Count() != 1 {
+		t.Fatalf("Unexpected count: %d", c.Count())
+	}
+
+	// Test delete.
+	c.Delete("key1")
+	v, ok = c.Get("key1")
+	if ok {
+		t.Fatalf("Unexpected value for key1 after delete: %v, %v", v, ok)
+	}
+
+	// Test clear.
+	c.Clear()
+	v, ok = c.Get("key10")
+	if ok || v != "" {
+		t.Fatalf("Unexpected value for key10 after clear: %v, %v", v, ok)
+	}
```

Comment:
`key10` never appears to be set in this test, so I think this is testing the `Clear` of an empty cache, which may not have been the intention.